### PR TITLE
fix(docs-infra): fix `search-results` styles

### PR DESCRIPTION
**Pull request by gkalpak**
I started to fix the "No results found" message being hidden, but ended up doing some clean up and other minor fixes (see individual commits for details).

WRT to the main fix ("No results found" being hidden):

**BEFORE**
search-not-found-before

**AFTER**
search-not-found-after

##
Both before and after the "No results found" is displayed correctly on the "Page Not Found" page:

page-not-found-before-and-after

---
**UPDATE:**

This PR also [fixes a bug][1] in `ScrollService` that caused test flakes ([example][2]). No idea why the flakes started appearing on this PR (which was just changing CSS styles 🤷‍♂).

[1]: https://github.com/angular/angular/commit/119dcd146188307f0296132536d8eb311fda96c4
[2]: https://circleci.com/gh/angular/angular/381649

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]

**Comment by sjtrimble** [comment]
@gkalpak 

A couple UX/UI recommendations:
1) Don't actually show the "I have nothing to show you" message if we don't have any suggestions OR even better (maybe handle it in another PR) show them a few items where they might want to spend some time on like the blog, the guide, etc.
Screen Shot 2019-07-04 at 12 32 16 PM

2) Can we do something to fix the cut off face image at the top? :)
Screen Shot 2019-07-04 at 12 32 04 PM

**Comment by gkalpak** [comment]
Thx for taking a look, @sjtrimble. I fixed the image placement.

Regarding the "no results" message, I wouldn't be opposed to showing something (although I assume it should rarely come up for the `page-not found` error page). It is not related to this PR though (I am just fixing the color of the message to be different from the background and thus visible). Feel free to open an issue (or a PR :wink:) about changing the message and we can discuss the details there.

**Comment by sjtrimble** [comment]
One more thing @gkalpak .. I'm still seeing the headers showing up a white and thereby transparent. Try seeing here: https://pr31390-119dcd1.ngbuilds.io/ngmodules

File: aio/src/styles/2-modules/_search-results.scss [hunk 1]
```diff
@@ -20,81 +18,84 @@ aio-search-results {
...
-        @include font-size(14);
-        color: $lightgray;
-        text-decoration: none;
-        font-weight: normal;
-        &:hover {
+        .search-result-item {
+          @include font-size(14);
+          color: $lightgray;
+          font-weight: normal;
+          &a {
```
Comment (sjtrimble) [inline aio/src/styles/2-modules/_search-results.scss]: I normally put a line break here so it's easier to read ... if you feel like it :)

**Review by sjtrimble — CHANGES_REQUESTED** [review]

File: aio/src/styles/2-modules/_search-results.scss (hunk 1 above)
Comment (gkalpak) [reply aio/src/styles/2-modules/_search-results.scss]: I was keeping inline with the convention in this file, which seems to be "no line break, if it's targeting the same element". Happy to go either way.

**Comment by gkalpak** [comment]
> I'm still seeing the headers showing up a white and thereby transparent.

Great catch! 😱 😱 😱
Fixed.

File: aio/src/styles/2-modules/_search-results.scss (hunk 1 above)
Comment (gkalpak) [reply aio/src/styles/2-modules/_search-results.scss]: I added the line breaks (since I agree it is more readable).

**Review by sjtrimble — APPROVED** [review]

**Pull request closed without merging**